docs/architecture/execution-engines.md

```diff
@@ -0,0 +1,42 @@
+## Execution Engines(?)
+
+A workload management system for the scheduling and running of jobs.  The typical enterprise requires support for multiple operating systems.  A core tenet of the project is to leverage both open source projects and commercial cloud products and their capabilities where we can. 
+TODO: Tie back to targeted use cases and typical enterprise
+
+Tier 1 supported environment is Linux and a typical use case of web serving and associated backend systems.  
+
+Tier 2 supported environment is Mac OSX for iOS and desktop clients.  We may also support Windows in the future.
+
+For tier 2 supported environments, we use Jenkins as our execution engine to handle scheduling across Mac OSX slaves.  Jenkins support across many OSes is useful here.
+
+For tier 1, we are evaluating multiple options for open source release.  While Jenkins serves us well in other areas, it has issues scaling and limits overall performance with its architecture.  In addition, managing a Jenkins cluster and the slaves has high operational overhead.
```

Comment:
Are slaves included in the Jenkins cluster? If so, I think it should either be `Jenkins master and slaves` or just `Jenkins cluster`